### ISSUE TITLE
linq2db doesn't work with MySqlConnector 2.0

### ISSUE THREAD

**Issue by klym1** [linq2db/linq2db#3341]
A few days ago a major update for MySqlConnector (https://github.com/mysql-net/MySqlConnector) was released (https://github.com/mysql-net/MySqlConnector/releases/tag/2.0.0)

Unfortunately, it does not work with latest linq2db (3.5.1). The exception is thrown when LinqToDB.Data.DataConnection is being instantiated:

```csharp
System.AggregateException: One or more errors occurred. (Method not found in target type: MySqlConnector.MySqlBulkCopy.WriteToServer(IDataReader))
 ---> LinqToDB.LinqToDBException: Method not found in target type: MySqlConnector.MySqlBulkCopy.WriteToServer(IDataReader)
   at LinqToDB.Expressions.TypeMapper.<>c.<ReplaceTypes>b__26_0(ReplaceTypesContext context, Expression e)
   at LinqToDB.Expressions.TransformVisitor`1.Transform(Expression expr)
   at LinqToDB.Expressions.Extensions.Transform[TContext](Expression expr, TContext context, Func`3 func)
   at LinqToDB.Expressions.TypeMapper.ReplaceTypes(Expression expression, ReplaceTypesContext ctx)
   at LinqToDB.Expressions.TypeMapper.MapLambdaInternal(LambdaExpression lambda, Boolean mapConvert, Boolean convertResult, Boolean ignoreMissingMembers)
   at LinqToDB.Expressions.TypeMapper.BuildWrapperImpl(LambdaExpression lambda, Boolean wrapResult, Boolean optional)
   at LinqToDB.Expressions.TypeMapper.BuildWrapper(LambdaExpression lambda, Boolean optional)
   at LinqToDB.Expressions.TypeMapper.<BuildWrapperMethods>b__17_1(<>f__AnonymousType114`2 e)
   at System.Linq.Utilities.<>c__DisplayClass2_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at LinqToDB.Expressions.TypeMapper.BuildWrapperMethods(Type wrapperType)
   at LinqToDB.Expressions.TypeMapper.FinalizeMappings()
   at LinqToDB.DataProvider.MySql.MySqlProviderAdapter.MySqlConnector.CreateAdapter()
   at LinqToDB.DataProvider.MySql.MySqlProviderAdapter.GetInstance(String name)
   at LinqToDB.DataProvider.MySql.MySqlDataProvider..ctor(String name, MappingSchema mappingSchema)
   at LinqToDB.DataProvider.MySql.MySqlDataProvider..ctor(String name)
   at LinqToDB.DataProvider.MySql.MySqlTools.<>c.<.cctor>b__18_1()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at LinqToDB.DataProvider.MySql.MySqlTools.GetDataProvider(String providerName)
   at LinqToDB.DataProvider.MySql.MySqlTools.ProviderDetector(IConnectionStringSettings css, String connectionString)
   at LinqToDB.Data.DataConnection.ConfigurationInfo.<>c__DisplayClass12_0.<GetDataProvider>b__0(Func`3 d)
   at System.Linq.Enumerable.SelectListIterator`2.MoveNext()
   at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.FirstOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
   at LinqToDB.Data.DataConnection.ConfigurationInfo.GetDataProvider(IConnectionStringSettings css, String connectionString)
   at LinqToDB.Data.DataConnection.GetDataProvider(String providerName, String connectionString)
   at LinqToDB.Data.DataConnection..ctor(LinqToDbConnectionOptions options)
   at ....LinqToDbConn`1..ctor(LinqToDbConnectionOptions connectionFactory)
```

### Environment details
linq2db version: *3.5.1*
Database Server: *MySql 5.7*
Database Provider: *MySqlConnector 2.0*
Operating system: *Windows 11*
.NET Framework: *.Net Core 5*

**Comment by MaceWindu** [comment]
Fix already released as part of [3.5.2 release](https://github.com/linq2db/linq2db/releases/tag/v3.5.2)

**Comment by klym1** [comment]
Great!
